hphp/runtime/base/variable-serializer.cpp

```diff
@@ -240,12 +240,15 @@ void VariableSerializer::write(double v) {
     if (!std::isinf(v) && !std::isnan(v)) {
       char *buf;
       vspprintf(&buf, 0, "%.*k", precision, v);
+      if (m_option & k_JSON_PRESERVE_ZERO_FRACTION
+          && strchr(buf, '.') == nullptr) {
+        free(buf);
+        asprintf(&buf, "%.1f", v);
```

Comment:
Won't this disregard `precision`? Also, is `asprintf` locale-independent, i.e. will this also use `.` instead of `,`?

I think it would be safer to just do the same as in PHP and do an extra append of `".0"`.